business/istio_status.go

```diff
@@ -251,6 +261,30 @@ func (iss *IstioStatusService) getAddonComponentStatus() IstioComponentStatus {
 	return ics
 }
 
+func (iss *IstioStatusService) getIstiodReachingCheck() (IstioComponentStatus, error) {
+	cfg := config.Get()
+	ics := IstioComponentStatus{}
+
+	istiods, err := iss.k8s.GetPods(cfg.IstioNamespace, labels.Set(map[string]string{"app": "istiod"}).String())
+	if err != nil {
+		return ics, err
+	}
+
+	for _, istiod := range istiods {
+		_, err := iss.k8s.GetPodProxy(istiod.Namespace, istiod.Name, "/ready")
```

Comment:
Just asking: wouldn't it be equivalent to get just the pod status? (ready/notready/ ...) hence relying on k8s API rather than crafting a custom query? Or this is not equivalent?

Reply:
Here is the main point. If we only rely on the k8s API we can't detect whether the cluster allow traffic between k8s API server and istiods.

We need this because when we fetch the proxy status, we connect to each istiod pod to fetch its `/debug/syncz` endpoint. This connection is done via k8s API: `kiali -> k8s API (proxy)-> istiod`.

The issue with private clusters usually has a really restricted K8S namespace access. Therefore, this check allows us to know whether the permissions are in place or not.

Reply:
A simple one-line comment in the code is useful when there is a design decision that may not be clear to the next dev.  You put thought into the approach, share that thought in the code when it may not be obvious.